meson.build

```diff
@@ -19,66 +18,97 @@ compiler_args = [
   '-Wno-microsoft-exception-spec',
 ]
 
-link_args = [
-  '-static',
-  '-static-libgcc',
-  '-static-libstdc++',
-  # We need to set the section alignment for debug symbols to
-  # work properly as well as avoiding a memcpy from the Wine loader.
-  '-Wl,--file-alignment=4096',
-]
-
-# Wine's built-in back traces only work with dwarf2 symbols
-if get_option('debug') and target_machine.system() == 'windows'
-  compiler_args += [
-    '-gstrict-dwarf',
-    '-gdwarf-2',
-  ]
-endif
+link_args = []
 
 if get_option('build_id')
   link_args += [
     '-Wl,--build-id',
   ]
 endif
 
-if cpu_family == 'x86'
+if platform == 'windows'
+  compiler_args += [
+    '-DNOMINMAX',
+    '-D_WIN32_WINNT=0xa00',
+  ]
+
   link_args += [
-    '-Wl,--enable-stdcall-fixup',
-    '-Wl,--add-stdcall-alias',
+    '-static',
+    '-static-libgcc',
+    '-static-libstdc++',
+    # We need to set the section alignment for debug symbols to
+    # work properly as well as avoiding a memcpy from the Wine loader.
+    '-Wl,--file-alignment=4096',
+  ]
+
+  # Wine's built-in back traces only work with dwarf2 symbols
+  if get_option('debug')
+    compiler_args += [
+      '-gstrict-dwarf',
+      '-gdwarf-2',
+    ]
+  endif
+
+  # Enable stdcall fixup on 32-bit
+  if cpu_family == 'x86'
+    link_args += [
+      '-Wl,--enable-stdcall-fixup',
+      '-Wl,--add-stdcall-alias',
   ]
+  endif
+
+  if (cpu_family == 'x86_64')
+    dxvk_library_path = meson.current_source_dir() + '/lib'
+  else
+    dxvk_library_path = meson.current_source_dir() + '/lib32'
+  endif
+
+  lib_vulkan  = cpp.find_library('vulkan-1', dirs : dxvk_library_path)
+  lib_d3d9    = cpp.find_library('d3d9')
+  lib_d3d11   = cpp.find_library('d3d11')
+  lib_dxgi    = cpp.find_library('dxgi')
+  lib_d3dcompiler_43 = cpp.find_library('d3dcompiler_43', dirs : dxvk_library_path)
+
+  if dxvk_is_msvc
+    lib_d3dcompiler_47 = cpp.find_library('d3dcompiler')
+  else
+    lib_d3dcompiler_47 = cpp.find_library('d3dcompiler_47')
+  endif
+
+  if dxvk_is_msvc
+    res_ext = '.res'
+    wrc = find_program('rc')
+    wrc_generator = generator(wrc,
+      output    : [ '@BASENAME@' + res_ext ],
+      arguments : [ '/fo', '@OUTPUT@', '@INPUT@' ],
+    )
+  else
+    res_ext = '.o'
+    wrc = find_program('windres')
+    wrc_generator = generator(wrc,
+      output    : [ '@BASENAME@' + res_ext ],
+      arguments : [ '-i', '@INPUT@', '-o', '@OUTPUT@' ],
+    )
+  endif
+
+  dxvk_include_path = include_directories('./include')
+else
+  lib_vulkan = cpp.find_library('vulkan')
+  lib_sdl2   = cpp.find_library('SDL2')
+
+  wrc           = find_program('touch')
+  wrc_generator = generator(wrc, output : [ '@BASENAME@_ignored.h' ], arguments : [ '@OUTPUT@' ] )
```

Comment:
shouldn't we stop using wrc unconditionally instead?

Reply:
There is a thing in meson called ["disabler"](https://mesonbuild.com/Disabler.html), but idk if it can help here, will it disable only generator or a whole target too.

Reply:
Its purpose is to cascade, it will convert all `wrc_generator.process()` to assign disablers instead of processed files, and consequently all `shared_library(..., foo_res)` are converted to disablers as well.

The typical use case is where, for example, an optional component of a build has a hard dependency on some dependency, and if that dependency cannot be found, the optional component shouldn't be built at all.

Reply:
I think the point here is that we should conditionally execute the generator and only the resulting files to the respective file list if we're actually building for Windows.

Reply:
My vote for leaving this stub as is (at least for now). Otherwise you'll likely end up adding even more stubs in every target which uses wrc_generator.